.github/workflows/integration_testing.yml

```diff
@@ -0,0 +1,81 @@
+name: Integration Testing
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-latest
+
+    env:
+      FASTN_DB_URL: postgres://testuser:testpassword@localhost:5432/testdb
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: testdb
+          POSTGRES_USER: testuser
+          POSTGRES_PASSWORD: testpassword
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          override: true
+          components: rustfmt, clippy
+
+      - name: fastn installation (using source)
+        id: fastn-install
+        continue-on-error: true
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: --path fastn --profile release
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install Python Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install Flask
+          pip install psycopg2
+
+      - name: Wait for PostgreSQL to start
+        run: |
+          timeout=30
+          until pg_isready -h localhost -p 5432 -U testuser -d testdb || ((timeout-- <= 0)); do
+            sleep 1
+          done
+
+      - name: Populate sample data
+        run: |
+          python scripts/populate-table.py
+
+      - name: Test server (port 5000)
+        run: |
+          python scripts/test-server.py &
+          
+          # Waiting for the server to start
+          sleep 10  # We might want to wait more
+
+      - name: Run integration tests
+        run: |
+          cd integration-tests
+          fastn test --headless
```

Comment:
Move all these to a single script: `scripts/run-intergration-test.sh`. Set -e in the script so we see when it fails.

Reply:
`set -eou pipefail`

Reply:
`set -o xtrace`